File: packages/python/convai_analytics/resources/usage.py
"""Usage facade — account/character/experience usage analytics."""

from __future__ import annotations

from typing import TYPE_CHECKING, Any

from ..measures import GroupBy, Measures
from ..types import BreakdownResponse

if TYPE_CHECKING:
    from ..client import ConvaiAnalytics


class UsageFacade:
    def __init__(self, client: ConvaiAnalytics) -> None:
        self._client = client

    def summary(
        self,
        *,
        group_by: str = GroupBy.CHARACTER_ID,
        **filters: Any,
    ) -> BreakdownResponse:
        """"How many sessions / unique end users / interactions did I have, by character?"

        Delegates to ``breakdown(measure='uniqueSessions', group_by=...)``.
        """
        return self._client.breakdown(
            measure=Measures.UNIQUE_SESSIONS,
            group_by=group_by,
            **filters,
        )

    def interactions(
        self,
        *,
        group_by: str = GroupBy.CHARACTER_ID,
        **filters: Any,
    ) -> BreakdownResponse:
        """"How many interactions per character?" """
        return self._client.breakdown(
            measure=Measures.UNIQUE_TURNS,
            group_by=group_by,
            **filters,
        )

File: packages/typescript/src/resources/usage.ts
/**
 * UsageFacade — account/character/experience usage analytics.
 */

import type { ConvaiAnalytics } from "../client.js";
import { GROUP_BY, MEASURES } from "../measures.js";
import type { BreakdownResponse, CommonFilters, TimeRange } from "../types.js";

export interface UsageSummaryParams extends TimeRange, CommonFilters {
  /** What to group by. Default 'characterId'. Other useful: 'experienceId', 'appKey'. */
  groupBy?: "characterId" | "experienceId" | "appKey";
}

export class UsageFacade {
  readonly #client: ConvaiAnalytics;

  constructor(client: ConvaiAnalytics) {
    this.#client = client;
  }

  /**
   * "How many sessions / unique end users / interactions did I have, by character?"
   * Delegates to `breakdown(measure=uniqueSessions, groupBy=...)`.
   */
  summary(params: UsageSummaryParams = {}): Promise<BreakdownResponse> {
    const { groupBy = GROUP_BY.characterId, ...rest } = params;
    return this.#client.breakdown({
      measure: MEASURES.uniqueSessions,
      groupBy,
      ...rest,
    });
  }

  /** "How many interactions per character?" */
  interactions(params: UsageSummaryParams = {}): Promise<BreakdownResponse> {
    const { groupBy = GROUP_BY.characterId, ...rest } = params;
    return this.#client.breakdown({
      measure: MEASURES.uniqueTurns,
      groupBy,
      ...rest,
    });
  }
}
